### PR TITLE
Youtube music fetching playlist + linter

### PR DESCRIPTION
**Pull request by adrhem**
- I used the existing spotify code to fetch the playlist from youtube (music) #4 and #5 . 
- I also installed eslint and I configured it with airbnb setup.

LOTS OF CHANGES because _eslint_.

### CODE REVIEW

File: src/YoutubeView.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import logo from './assets/logo.png';
+import './styles/App.css';
+import { useYoutubeAuth } from './contexts/YoutubeContext';
+import { generateYoutubeUrl } from './services/Youtube';
+import { Playlist } from './types/Playlist';
+import { PlaylistQuery } from './types/PlaylistQuery';
+import { TOKEN_YOUTUBE_TYPE, TokenMessage } from './types/TokenTypes';
+import OpenAuthenticationPopup from './utils/openAuthenticationPopup';
+
+function YoutubeView(): JSX.Element {
```
Comment (elCarlosHez) [inline src/YoutubeView.tsx]:
```suggestion
export const YoutubeView = (): JSX.Element => {
```

I consider more cleaner to use arrow functions

File: src/YoutubeView.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      if (event.origin !== import.meta.env.VITE_APP_URL as string) return;
+
+      const message:TokenMessage = event.data;
+      // We check if the type we'll recieve is the type Spotify
+      if (message?.type === TOKEN_YOUTUBE_TYPE) {
+        setToken(message.token);
+      }
+    });
+    const url = generateYoutubeUrl();
+    OpenAuthenticationPopup(url, 'Spotify');
```
Comment (elCarlosHez) [inline src/YoutubeView.tsx]:
```suggestion
    OpenAuthenticationPopup(url, 'Youtube');
```

Typo error

File: src/contexts/SpotifyContext.tsx [hunk 3]
```diff
@@ -12,28 +12,28 @@ interface ISpotifyProvider {
   children: JSX.Element;
 }
 
-export const SpotifyProvider = (props: ISpotifyProvider) => {
+export function SpotifyProvider(props: ISpotifyProvider) {
```
Comment (elCarlosHez) [inline src/contexts/SpotifyContext.tsx]:
```suggestion
export const SpotifyProvider = (props: ISpotifyProvider): JSX.Element => {
```

File: src/YoutubeView.tsx (hunk 2 above)
Comment (adrhem) [reply src/YoutubeView.tsx]: lol

File: src/contexts/SpotifyContext.tsx [hunk 4]
```diff
@@ -12,28 +12,28 @@ interface ISpotifyProvider {
...
     } catch (error) {
-      console.log('Spotify fetch Error: ', error);
       return null;
     }
-  };
+  }
 
   return (
     <SpotifyContext.Provider
+      // eslint-disable-next-line react/jsx-no-constructed-context-values
```
Comment (elCarlosHez) [inline src/contexts/SpotifyContext.tsx:38]: I'm gonna check this error

File: src/contexts/YoutubeContext.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    try {
+      const promise = await fetch(`${url}?part=snippet&mine=true`, {
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+      });
+      return await promise.json();
+    } catch (error) {
```
Comment (elCarlosHez) [inline src/contexts/YoutubeContext.tsx:29]: Could you add a `console.log` with the error? So in case this fails, we could get the context

**Review by elCarlosHez — APPROVED** [review]
LGTM

Left a couple of suggestions

**Pull request merged**